llvm/lib/Transforms/Instrumentation/MemProfUse.cpp

```diff
@@ -752,3 +773,58 @@ PreservedAnalyses MemProfUsePass::run(Module &M, ModuleAnalysisManager &AM) {
 
   return PreservedAnalyses::none();
 }
+
+bool MemProfUsePass::annotateGlobalVariables(
+    Module &M, const memprof::DataAccessProfData *DataAccessProf) {
+  if (!AnnotationStaticDataPrefix || M.globals().empty() || !DataAccessProf)
```

Comment:
Can we emit a warning if the user wanted to annotate the static data but the memprof didn't have the info? It would catch mismatched expectations during a build.